readers-frontend/src/components/utils/articles.ts

```diff
@@ -120,6 +120,50 @@ export const getAllArticles = async (): Promise<articleWithUserAndDraft[]> => {
   return articles;
 };
 
+/**
+ * Gets 4 article with same parent category as the passed one.
+ * @param {number} parentCategoryId The parent category id.
+ * @returns {Promise<articlesWithArticleDrafts[]>} Promise object represents the articles.
+ */
+export const getSimilarArticles = async (
+  parentCategoryId: number
+): Promise<articlesWithArticleDrafts[]> => {
+  return await prisma.articles.findMany({
+    where: {
+      categories: {
```

Comment:
I would suggest excluding the main article id if possible. It would save you the hassle of having to filter the similar articles later

Reply:
> I would suggest excluding the main article id if possible. It would save you the hassle of having to filter the similar articles later

No cause this is run once per category. So if you exclude the one article, the similar articles for that page will only have 3 entries. Yes ideally you'd exclude the article here, bu there you'd have to perform the query for every article, and the build takes too long. Filtering a 5 element array is always gonna be quicker than perform the query for each article. Unless I misunderstood what you mean